src/main/java/com/minecolonies/core/entity/ai/workers/production/EntityAIWorkLumberjack.java

```diff
@@ -566,15 +579,6 @@ private IAIState chopTree()
             job.getTree().pollNextLog();
             worker.decreaseSaturationForContinuousAction();
         }
-        else if (job.getTree().hasLeaves() && shouldBreakLeaves)
```

Comment:
moving this to above will prevent leaf decay/mods like fast leaf decay so he'll spend much more time harvesting leaves

Reply:
That's true, but only when the setting is enabled, and it's usually disabled unless you're not entirely trusting the decay mechanism to drop the saplings quickly enough.

I did consider making it switch the orders only for the mushroom trees and not regular trees, but that seemed less consistent.  And requires a more awkward code structure.

Reply:
is there a reason why mushroom trres need to harvest them first?

Reply:
Mostly just that it looks really really silly to chop the trunk first and leave a huge mushroom head entirely unsupported while slowly chipping away at it.

Reply:
ye makes sense, wonder if normal trees need the leaf break at all then, maybe we just do it for mushroom ones

Reply:
As noted, there's a setting for normal trees.  The setting defaults to off, but people can turn it on if they like (e.g. they don't have fast decay and they don't think they're getting enough saplings, especially for things like dark oak).

For overworld/nether mushroom trees, it always breaks "leaves" regardless of the setting, because they don't otherwise decay.